Google Cloud Storage support

littleq0903 opened the pull request:
manual QA passed.

now adding unit testing on it.

winhamwr commented:
Hi Colin,

It also looks like you have some flake8 issues to clear up, if you didn't notice that build fail.

-Wes

littleq0903 commented:
I checked with PyFlake from my side, didn't see any error in the code I added, but found two additional errors in the beginning of tester classes, now fixed.

but from the message it seems like the error refers to the command-level error? if pep8 error it should point out which line.

littleq0903 commented:
oh wait, I checked the wrong files. let me fix it.

littleq0903 commented:
figured it out, I recently made my PyFlake vim plugin to ignore E501 errors (line too long exceed 80 chars per line), fixed all lines exceed the 80 chars.

littleq0903 commented:
now this left:

`ERROR: InvocationError: '/home/travis/build/PolicyStat/terrarium/.tox/py26-venv-1.8.4/bin/nosetests -v'`

winhamwr commented on terrarium/terrarium.py:
```diff
@@ -760,6 +844,48 @@ def parse_args():
+    )
+    ap.add_argument(
+        '--gcs-client-email',
+        default=os.environ.get('GCS_CLINET_EMAIL', None),
```
Typo here. `GCS_CLIENT_EMAIL`

winhamwr commented:
Other than the typo, I think this needs:

* A unittest like [this one](https://github.com/PolicyStat/terrarium/blob/master/tests/tests.py#L517) that verifies the error message/code if GCS isn't installed
* GCS documentation that mirrors [the s3 docs](https://github.com/PolicyStat/terrarium/blob/master/docs/user_guide.rst#storing-terrarium-environments-on-s3) to let folks know they can use GCS.

It looks like your last failing travis build might have been just intermittent. I made a small change to master just to confirm that the travis build is passing there and it is.

littleq0903 commented:
@winhamwr : can you re-run the build#199 on Travis?

winhamwr commented:
Hi Colin, 

Maybe fix the typo and push? I'm on my mobile and don't see an obvious way to do that. 

-Wes

littleq0903 commented:
docs added.

winhamwr commented:
Excellent. Thanks, Colin!
